agents/ocf/remote.in

```diff
@@ -28,11 +28,11 @@ meta_data() {
   <version>1.1</version>
   <shortdesc lang="en">Pacemaker Remote connection</shortdesc>
   <parameters>
-    <parameter name="server" unique-group="address">
+    <parameter name="addr" unique-group="address">
```

Comment:
Bah, I never noticed this discrepancy. "server" is what users have been seeing (from the metadata, which is used by both pcs and the Pacemaker Remote document). We have 10 regression tests using "server" and none using "addr" as a result. Unfortunately we can't change the "addr" entries added by Pacemaker without changing the config hash.

Also unfortunately, they're not exactly equivalent. pe__bundle_needs_remote_name(), pcmk__substitute_remote_addr(), and add_replica_actions_to_graph() via XPATH_REMOTE, only check "addr". (Using an XPath to search for a meta-attribute is problematic in general due to the possibility of multiple blocks using rules. That wouldn't make sense for this particular meta-attribute, but strange configs happen.) At least we could fix these spots to check for both.

I tend to think we should keep "server" since it is the only one that has been documented all this time. We could maybe change "addr" at 3.0.0.

Reply:
> Unfortunately we can't change the "addr" entries added by Pacemaker without changing the config hash.

Yep that was the reason I chose "addr" over "server".

> Also unfortunately, they're not exactly equivalent. pe__bundle_needs_remote_name(), pcmk__substitute_remote_addr(), and add_replica_actions_to_graph() via XPATH_REMOTE, only check "addr".

I don't think this matters. From a skim, those functions are looking for "#uname" as a special value of "addr", and only Pacemaker would ever set that. If a user sets `addr="#uname"`, that's a grave mistake and a nonsensical configuration AFAIK.

Also, if there is a situation where the user sets "server=<some_address>" and (for some reason) Pacemaker sets "addr" to "#uname" and then substitutes a calculated address later, the result in the RA would be undefined. The last one in the param list ("addr" or "server") takes precedence in `handle_remote_ra_start()`.

> we could fix these spots to check for both

No need, checking for `addr="#uname"` is checking for a value that Pacemaker planted there as part of the remote container hack. We only need to check whichever variable Pacemaker uses for the hack.

> I tend to think we should keep "server" since it is the only one that has been documented all this time. We could maybe change "addr" at 3.0.0.

We also use "remote-addr" (not "remote-server") as a guest node resource meta-attribute. In that case, we pass its value as the "addr" (not "server") of the implicit remote connection resource (via `pe_create_remote_xml()`).

The remote RA metadata is the **only** place we use "server". Deprecating it in favor of "addr" seems reasonable to me.

Waiting for 3.0.0 and then going the other direction -- changing all uses to "server" instead of changing the metadata to "addr" -- is still an option too.

Reply:
> The remote RA metadata is the **only** place we use "server".

Sure, but that's the documentation, so that's what users have been using

> Waiting for 3.0.0 and then going the other direction -- changing all uses to "server" instead of changing the metadata to "addr" -- is still an option too.

I'm thinking that keeping the current situation is the least disruptive.

Reply:
[T743](https://projects.clusterlabs.org/T743)